src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileRepository.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class FileRepository : IFileRepository
+    {
+        private readonly ContentDbContext _contentDbContext;
+
+        public FileRepository(ContentDbContext contentDbContext)
+        {
+            _contentDbContext = contentDbContext;
+        }
+
+        public async Task<ReleaseFileReference> Create(Guid releaseId,
+            string filename,
+            ReleaseFileTypes type,
+            ReleaseFileReference replacingFile = null,
+            ReleaseFileReference source = null)
+        {
+            if (type == ReleaseFileTypes.DataZip)
+            {
+                throw new ArgumentException($"Cannot use generic Create method for type {ReleaseFileTypes.DataZip}",
+                    nameof(type));
+            }
+
+            var releaseFile = new ReleaseFile
+            {
+                ReleaseId = releaseId,
+                ReleaseFileReference = new ReleaseFileReference
+                {
+                    ReleaseId = releaseId,
+                    Filename = filename,
+                    ReleaseFileType = type,
+                    Replacing = replacingFile,
+                    Source = source
+                }
+            };
+
+            var created = (await _contentDbContext.ReleaseFiles.AddAsync(releaseFile)).Entity;
+
+            if (replacingFile != null)
+            {
+                _contentDbContext.Update(replacingFile);
+                replacingFile.ReplacedBy = releaseFile.ReleaseFileReference;
+            }
+
```

Comment:
Shouldn't we save the context here? Same for the create, update and delete methods

Reply:
See what I put for the other answer and let me know what you think. 🙂